### ISSUE TITLE
Heading that contains {{DataValue}} produces invalid HTML/JS for ToC

### ISSUE THREAD

**Issue by marcelgerber** [owid/owid-grapher#1256]
## Description

With [Outdoor air pollution](https://ourworldindata.org/outdoor-air-pollution), we have a page that uses the `{{DataValue}}` syntax in a h3 heading, namely in `Outdoor air pollution is attributed to {{DataValue variableId:244614 entityId:355 chartId:1588 | %value}} each year`.

To produce the ToC, our site code produces the following `runTableOfContents` code and puts it into a `<script>` tag:
```js
runTableOfContents({
  headings: [
    { text: "Summary", slug: "summary", isSubheading: false },
    {
      text: "Outdoor air pollution is one of the leading risk factors for premature death",
      slug: "outdoor-air-pollution-is-one-of-the-leading-risk-factors-for-premature-death",
      isSubheading: false,
    },
    {
      text: "Outdoor air pollution is attributed to 4.51 million deaths each year",
      // ⬇️⬇️⬇️
      html: 'Outdoor air pollution is attributed to <span class="annotating-data-value" data-reactroot><script data-type="AnnotatingDataValue" type="component/props">{"value":4506193,"formattedValue":"4.51 million deaths","template":"%value","year":2019,"unit":"deaths","entityName":"World"}</script><span class><span class="data-value">4.51 million deaths</span></span></span> each year',
      // ⬆️⬆️⬆️
      slug: "outdoor-air-pollution-is-attributed-to-4-51-million-deaths-each-year",
      isSubheading: true,
    },
    // ...
  ],
  pageTitle: "Outdoor Air Pollution",
});
```

As you can see, that second `html` property contains a `<script><script/>` tag, **and the HTML parser will parse the `</script>` as the end of the entire "big" script block** (containing the `runTableOfContents` call).

That's why, when you navigate to the page, you'll see a `SyntaxError: Invalid or unexpected token` error in the console.

## Effects

This has the following effects:
* The ToC doesn't work (it doesn't even open)
* Related charts are not shown
* At the very bottom of the page, there's some extra JS
	  <details>
	  <summary>Details</summary>
	  ![image](https://user-images.githubusercontent.com/2641501/158080582-da9e6fd1-b77c-469e-9d45-8df391165619.png)
	  </details>
* However, all charts are interactive

## Possible solutions

* [There's a blog post](https://remarkablemark.org/blog/2019/04/29/include-script-inside-script-tag/) that outlines how a `</script>` tag can be safely included **inside a string** in a `</script>` tag.
    tl;dr: basically just `html.replace(/<script\b/g, "\\x3Cscript")`
* Disallow `{{DataValue}}` in headings, and enforce it
* Don't use a `<script>` element to store the data value

## Additional context

Found via Bugsnag here: https://app.bugsnag.com/our-world-in-data/our-world-in-data-website/errors/61f274eef420a200099a5726

It appears that `outdoor-air-pollution` is the only page where this error occurs.

**Comment by larsyencken** [comment]
Marking it as a 1h fix, I'd be tempted for us to just ship the `</script>` replacement regex to sanitise the table of contents `html` blocks. If we can't do it quick and easy, then we should not do it at all and just change the title for this page.

**Comment by mlbrgl** [comment]
Quite a specific use case, thanks for picking that up @marcelgerber! 

As a reminder, `{{DataValue}}` are still [considered experimental](https://www.notion.so/DataValue-tag-5ed2ab3f350048168758aecc19ff9cad).

**Comment by mlbrgl** [comment]
I also meant to say, thanks for the very detailed bug report 👍
